File: C2/PDOO/Practicas/DeepSpaceJava/test/TestP4.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author JuanAntonio
 */
public class TestP4 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        
        SuppliesPackage sp=new SuppliesPackage(1,10,1);
        SpaceStation station = new SpaceStation("Juan",sp);
        //System.out.println(station.toString());
        
        //gameUniverse.init(nombres); Revienta porque CardDealer tiene errores
        //System.out.println(station.fire());
        //station = new PowerEfficientSpaceStation(station);
        //System.out.println(station.toString());        
        station.receiveHangar(new Hangar(3));
        station.receiveShieldBooster(new ShieldBooster("Escudo ACME",1.5f,2));
        station.receiveShieldBooster(new ShieldBooster("Escudo ACME",1.5f,2));
        station.receiveWeapon(new Weapon("Misil ACME",WeaponType.MISSILE,2));
        station.receiveWeapon(new Weapon("Cañón definitivo",WeaponType.PLASMA,100));
        station.mountShieldBooster(0);
        station.mountWeapon(0);
        
        station = new PowerEfficientSpaceStation(station);
        System.out.println(station.fire());
        System.out.println(station.protection());
        station = new BetaPowerEfficientSpaceStation(station);
        System.out.println(station.fire());
        System.out.println(station.protection());
        
        System.out.println(station.toString());
        
        ArrayList<SpaceStation> collaborators=new ArrayList<>();
        collaborators.add(station);
        SpaceStation station2=new SpaceStation("Pepe", sp);
        station2=new SpaceCity(station2, collaborators);
        System.out.println(station2.fire());
        System.out.println(station2.protection());
        
        System.out.println(station2.toString());
        
        ArrayList <WeaponType> wl=new ArrayList<>();
        wl.add(WeaponType.LASER);
        wl.add(WeaponType.MISSILE);
        wl.add(WeaponType.PLASMA);
        NumericDamage damage1=new NumericDamage(5, 4);
        SpecificDamage damage2=new SpecificDamage(wl, 4);
        
        System.out.println(damage1.toString());
        System.out.println(damage2.toString());
        
        System.out.println("===================================================================");
        
        Loot loot=new Loot(1,1,1,1,1,false, true);
        Loot loot2=new Loot(1,1,1,1,1,true, false);
        
        Transformation transformacion1=station.setLoot(loot);
        System.out.format("\n%s\n",transformacion1.toString());//No se trasforma=bien
        SpaceStation station3=new SpaceStation("Luis", sp);
        Transformation transformacion2=station3.setLoot(loot);
        System.out.format("\n%s\n",transformacion2.toString());//Se trasforma=bien, ya que el loot no transforma por si solo
        Transformation transformacion3=station3.setLoot(loot2);
        System.out.format("\n%s\n",transformacion3.toString());//Se trasforma=bien, ya que el loot no transforma por si solo
        Transformation transformacion4=station2.setLoot(loot);
        System.out.format("\n%s\n",transformacion4.toString());//No se trasforma=bien
        Transformation transformacion5=station2.setLoot(loot2);
        System.out.format("\n%s\n",transformacion5.toString());//No se trasforma=bien
        
        
        ArrayList<String> nombres=new ArrayList();
        nombres.add("Juan");
        nombres.add("Antonio");
        nombres.add("Pedro");
        GameUniverse gameUniverse = new GameUniverse();
        gameUniverse.init(nombres);
        gameUniverse.makeStationEfficient();
        //gameUniverse.createSpaceCity();
        System.out.format("\n%s\n",gameUniverse.toString());
        
    }
    
}

File: C2/PDOO/Practicas/DeepSpaceRuby/test/test_p4.rb
#encoding:utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "SpecificDamage"
require_relative "Damage"
require_relative "Hangar"
require_relative "Weapon"
require_relative "GameUniverse"
require_relative "EnemyStarShip"
require_relative "SpaceStation"
require_relative "SpaceCity"

module Deepspace
  class TestP4
    def self.main     
      weapons=[]
      #weapons.push(Weapon.new("Láser infinito",WeaponType::LASER,100))
      #weapons.push(Weapon.new("Láser ACME",WeaponType::LASER,1))
      weapons.push(Weapon.new("Láser ACME",WeaponType::LASER,1))
      #weapons.push(Weapon.new("Misil ACME",WeaponType::MISSILE,1))
      weapons.push(Weapon.new("Cañón ACME",WeaponType::PLASMA,1)) 
      
      shieldBoosters=[]
      shieldBoosters.push(ShieldBooster.new("El escudo infinito",2.0,100))
      #shieldBoosters.push(ShieldBooster.new("El escudo definitivo",10.0,2))
      shieldBoosters.push(ShieldBooster.new("Escudo ACME",1.5,2))
      
      
      damage=SpecificDamage.new([WeaponType::LASER,WeaponType::MISSILE,WeaponType::MISSILE,WeaponType::MISSILE,WeaponType::PLASMA],4)
      
      puts damage.to_s
      damage=damage.adjust(weapons, shieldBoosters)
      puts damage.to_s
      
      #damage1=damage1.adjust(weapons, shieldBoosters)
      #puts damage2.to_s
      
      suppliesPackages=[]
      suppliesPackages.push(SuppliesPackage.new(1,30,2))
      suppliesPackages.push(SuppliesPackage.new(3,0,1))
      suppliesPackages.push(SuppliesPackage.new(1,0,3))
      
      collaborators=[]
      collaborators.push(SpaceStation.new("Juan",suppliesPackages[0]))
      collaborators.push(SpaceStation.new("Antonio",suppliesPackages[1]))

      
    

      
      transformLoot1=Loot.new(1,1,1,1,1,true,false)
=begin      
      collaborators.each{ |estacion|
        estacion.setLoot(goodLoot)
      }
      base.setLoot(transformLoot1)
      city=SpaceCity.new(base,collaborators)
      puts city.to_s
      city.mountWeapon(0)
      city.mountShieldBooster(0)
      puts city.fire
      puts city.protection
      
=end      puts city.to_s    
      gameUniverse=GameUniverse.new
      nombres=["Juan", "Antonio"]
      gameUniverse.init(nombres)
      #puts gameUniverse.to_s
      #gameUniverse.createSpaceCity
      #puts gameUniverse.to_s
     
=begin      transformLoot0=Loot.new(1,1,1,0,1,true,false)
      base=SpaceStation.new("BASE",suppliesPackages[2])
      if base.setLoot(transformLoot0)==Transformation::GETEFFICIENT
        base=BetaPowerEfficientSpaceStation.new(base)
      end
      puts base.to_s
=end
      
      enemy=EnemyStarShip.new("Malo",1,2,transformLoot1, NumericDamage.new(1,1))
      station=SpaceStation.new("Juan", SuppliesPackage.new(500,50,500))
      
      puts gameUniverse.combatGo(station, enemy)
      puts gameUniverse.to_s
      
      
    end
    
    
    
  end
  
TestP4.main  
 
end
